### PR TITLE
Delete old nbr sampling software

### PR DESCRIPTION
**Pull request by ChuckHastings**
Time ran out during 22.06 development to replace the old code.

This PR renames the experimental version of uniform neighborhood sampling to the regular name and eliminates the original implementation.

Tagged as `breaking` since we eliminate a function from the C++ API.

### CODE REVIEW

**Review by seunghwak — APPROVED** [review]

**Comment by BradReesWork** [comment]
rerun tests

File: cpp/src/sampling/detail/sampling_utils_sg.cu [hunk 1]
```diff
@@ -84,6 +85,7 @@ template rmm::device_uvector<int64_t> get_global_adjacency_offset(
   graph_view_t<int64_t, int64_t, double, false, false> const& graph_view,
   rmm::device_uvector<int64_t> const& global_degree_offsets,
   rmm::device_uvector<int64_t> const& global_out_degrees);
+#endif
```
Comment (seunghwak) [inline cpp/src/sampling/detail/sampling_utils_sg.cu:88]: Better just delete?

**Review by seunghwak — APPROVED** [review]

File: cpp/src/sampling/detail/sampling_utils_sg.cu (hunk 1 above)
Comment (seunghwak) [reply cpp/src/sampling/detail/sampling_utils_sg.cu:88]: And can I just close PR https://github.com/rapidsai/cugraph/pull/2370?

Comment (ChuckHastings) [reply cpp/src/sampling/detail/sampling_utils_sg.cu]: I think you can just close PR 2370.

I deleted both of these blocks of code (in _sg and _mg).  Thanks for catching that.

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**